File: 982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.cpp
class Solution {
public:
    int minIncrementForUnique(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int t = nums[0]+1;
        int ans = 0;
        for(int i=1;i<nums.size();i++){
            if(t>nums[i]){
                ans+=t-nums[i];
                t++;
            }
            else{
                t=nums[i]+1;
            }
        }
        return ans;
    }
};

File: 982-minimum-increment-to-make-array-unique/minimum-increment-to-make-array-unique.java
class Solution {
    public int minIncrementForUnique(int[] nums) {
        Arrays.sort(nums);
        int target = nums[0];  // Initialize target to nums[0]
        int ans = 0;
        for (int i = 0; i < nums.length; i++) {
            if (target > nums[i]) {
                ans += target - nums[i];
            } else {
                target = nums[i];
            }
            target++;  // Increment target for the next position
        }
        return ans;
    }
}
